lisa/sut_orchestrator/qemu/platform.py

```diff
@@ -119,8 +172,64 @@ def _create_node_capabilities(
         node_capabilities.set_extended_runbook(
             node_space.get_extended_runbook(QemuNodeSchema, type_name=QEMU)
         )
+
         return node_capabilities
 
+    # Check that the VM requirements can be fulfilled by the host.
+    def _check_host_capabilities(
+        self,
+        nodes_requirements: List[schema.NodeSpace],
+        host_capabilities: _HostCapabilities,
+        log: Logger,
+    ) -> bool:
+        total_required_memory_mib = 0
+
+        for node_requirements in nodes_requirements:
+            # Ensure host has enough CPU cores for the VM.
+            # Note: The CPU scheduler can easily handle overprovisioning of CPU
+            # cores.
+            assert isinstance(node_requirements.core_count, int)
+            if node_requirements.core_count > host_capabilities.core_count:
+                log.error(
+                    f"Node requires {node_requirements.core_count} CPU cores. "
+                    f"Host only has {host_capabilities.core_count}."
+                )
+                return False
+
+            # Calculate the total amount of memory required for all the VMs.
+            assert isinstance(node_requirements.memory_mb, int)
+            total_required_memory_mib += node_requirements.memory_mb
+
+        # Ensure host has enough memory for all the VMs.
+        total_required_memory_kib = total_required_memory_mib * 1024
+        if total_required_memory_kib > host_capabilities.free_memory_kib:
+            log.error(
+                f"Nodes require a total of {total_required_memory_kib} KiB memory. "
+                f"Host only has {host_capabilities.free_memory_kib} KiB free."
+            )
+            return False
+
+        return True
+
+    # Get the minimum value for a node requirement with an interger type.
+    # Note: Unlike other orchestrators, we don't want to fill up the capacity of
+    # the host in case the test is running on a dev box.
+    def _get_count_space_min(self, count_space: search_space.CountSpace) -> int:
```

Comment:
It can use `generate_min_capability_countspace`, and put in the count_space twice. It covers all scenarios.